Make internal (core) errors show up in the Qt client.

zander opened the pull request:
Serious warnings wrt to state of the chain were only printed to the logs, with this change these issues are also shown in the Qt Gui..

luke-jr committed:
-bytespersigop option to additionally limit sigops in transactions we relay and mine

murchandamus committed:
Coinselection prunes extraneous inputs from ApproximateBestSubset

This is a combination of 3 commits.

- Coinselection prunes extraneous inputs from ApproximateBestSubset
  A further pass over the available inputs has been added to ApproximateBestSubset after a candidate set has been found. It will prune any extraneous inputs in the selected subset, in order to decrease the number of input and the resulting change.
- Moved set reduction to the end of ApproximateBestSubset to reduce performance impact
- Added a test for the pruning of extraneous inputs after ApproximateBestSet

Github-Pull: #4906
Rebased-From: 5c03483e26ab414d22ef192691b2336c1bb3cb02 af9510e0374443b093d633a91c4f1f8bf5071292 fc0f52d78085b6ef97d6821fc7592326c2d9b495

sipa committed:
Replace trickle nodes with per-node/message Poisson delays

We used to have a trickle node, a node which was chosen in each iteration of
the send loop that was privileged and allowed to send out queued up non-time
critical messages. Since the removal of the fixed sleeps in the network code,
this resulted in fast and attackable treatment of such broadcasts.

This pull request changes the 3 remaining trickle use cases by random delays:
* Local address broadcast (while also removing the the wiping of the seen filter)
* Address relay
* Inv relay (for transactions; blocks are always relayed immediately)

The code is based on older commits by Patrick Strateman.

Github-Pull: #7125
Rebased-From: 5400ef6bcb9d243b2b21697775aa6491115420f3

jtimon committed:
TRIVIAL: Chainparams: Remove unused CBaseUnitTestParams

paveljanik committed:
PartitionCheck: remove useless spaces

luke-jr committed:
Constrain rpcport default values to a single location in code

paveljanik committed:
Fix various typos

antonio-fr committed:
Update miner.cpp: Fix typo in comment

Commit:
[trivial] Fix typo in peertablemodel.cpp

TheBlueMatt committed:
Add failing test checking timelocked-txn removal during reorg

TheBlueMatt committed:
Fix removal of time-locked transactions during reorg

TheBlueMatt committed:
Fix comment in removeForReorg

TheBlueMatt committed:
Make indentation in ActivateBestChainStep readable

TheBlueMatt committed:
removeForReorg calls once-per-disconnect-> once-per-reorg

Commit:
rpcwallet: Clarify what settxfee does

TheBlueMatt committed:
Add method to remove a tx from CCoinsViewCache if it is unchanged

TheBlueMatt committed:
Get the set of now-uncacheable-txn from CTxMemPool::TrimToSize

TheBlueMatt committed:
Discard txn cache entries that were loaded for removed mempool txn

TheBlueMatt committed:
Add CCoinsViewCache::HaveCoinsInCache to check if a tx is cached

TheBlueMatt committed:
Uncache input txn in utxo cache if a tx is not accepted to mempool

sipa committed:
Flush coins cache also after transaction processing

morcos committed:
Store the total sig op count of a tx.

Store sum of legacy and P2SH sig op counts.  This is calculated in AcceptToMemory pool and storing it saves redoing the expensive calculation in block template creation.

morcos committed:
Add a score index to the mempool.

The score index is meant to represent the order of priority for being included in a block for miners.  Initially this is set to the transactions modified (by any feeDelta) fee rate.  Index improvements and unit tests by sdaftuar.

morcos committed:
Make accessing mempool parents and children public

morcos committed:
Expose FormatStateMessage

morcos committed:
Change default block priority size to 0

Make RPC tests have a default block priority size of 50000 (the old default) so we can still use free transactions in RPC tests.  When priority is eliminated, we will have to make a different change if we want to continue allowing free txs.

sdaftuar committed:
Track coinbase spends in CTxMemPoolEntry

This allows us to optimize CTxMemPool::removeForReorg.

morcos committed:
Add TxPriority class and comparator

morcos committed:
Rewrite CreateNewBlock

Use the score index on the mempool to only add sorted txs in order.  Remove much of the validation while building the block, relying on mempool to be consistent and only contain txs that can be mined.
The mempool is assumed to be consistent as far as not containing txs which spend non-existent outputs or double spends, and scripts are valid.  Finality of txs is still checked (except not coinbase maturity, assumed in mempool).
Still TestBlockValidity in case mempool consistency breaks and return error state if an invalid block was created.
Unit tests are modified to realize that invalid blocks can now be constructed if the mempool breaks its consistency assumptions and also updated to have the right fees, since the cached value is now used for block construction.

Conflicts:
	src/miner.cpp

Commit:
HelpMessage: Don't hide -mintxfee behind showDebug

fanquake committed:
[Depends] Bump Boost, miniupnpc, ccache & zeromq

Bring dependencies up to date with master:
[depends] Boost 1.59.0
[depends] miniupnpc 1.9.20151026
[depends] native ccache 3.2.4
[depends] zeromq 4.0.7
[depends] Latest config.guess & config.sub
[depends] Fix miniupnpc compilation on osx

Github-Pull: #6980
Rebased-From: 9e940fa4c650dd31c39dbc8ed4038e131c19d59c 17ad964c2ff8f9be62a6826012b565843d3d72ba 26f8ea5342994bc3dcc22163b86f086328b50769 10d3c77644d894338a02b05f64ba822f3a516401 23a3c47f95c9c7c1778c488be6ea9ebbef2311ea e0769e1928f892fb16f851991d8e09a90587a1f4

sdaftuar committed:
Don't call removeForReorg if DisconnectTip fails

sdaftuar committed:
Add rounding helper function to util.py

sdaftuar committed:
Add rpc test for prioritisetransaction

laanwj committed:
build: Make networking work inside LXC builder in gitian-building.md

These are changes I needed to get gitian building to work with Debian
8.2, which is the version we tell to use.

- Set up NAT, so that container can access network beyond host
- Remove explicit cgroup setup - these are mounted automatically now
- gitian: Need `ca-certificates` and `python` for LXC builds

Github-Pull: #7060
Rebased-From: 99fda26de0661afcbe43d5e862c382e3c2e3aa5e 3b468a0e609147c7d7afd8ed97bf271f2356daef

sdaftuar committed:
[Mempool] Fix mempool limiting and replace-by-fee for PrioritiseTransaction

1) Fix mempool limiting for PrioritiseTransaction

Redo the feerate index to be based on mining score, rather than fee.

Update mempool_packages.py to test prioritisetransaction's effect on
package scores.

2) Update replace-by-fee logic to use fee deltas

3) Use fee deltas for determining mempool acceptance

4) Remove GetMinRelayFee

One test in AcceptToMemoryPool was to compare a transaction's fee
agains the value returned by GetMinRelayFee. This value was zero for
all small transactions.  For larger transactions (between
DEFAULT_BLOCK_PRIORITY_SIZE and MAX_STANDARD_TX_SIZE), this function
was preventing low fee transactions from ever being accepted.

With this function removed, we will now allow transactions in that range
with fees (including modifications via PrioritiseTransaction) below
the minRelayTxFee, provided that they have sufficient priority.

Github-Pull: #7062
Rebased-From: eb306664e786ae43d539fde66f0fbe2a3e89d910 9ef2a25603c9ec4e44c4f45c6a5d4e4386ec86d3 27fae3484cdb21b0d24face833b966fce5926be5 901b01d674031f9aca717deeb372bafa160a24af

gmaxwell committed:
Decide eviction group ties based on time.

This corrects a bug the case of tying group size where the code may
 fail to select the group with the newest member. Since newest time
 is the final selection criteria, failing to break ties on it
 on the step before can undermine the final selection.

Tied netgroups are very common.

Commit:
[doc] Minor markdown fixes

sdaftuar committed:
Fix removeForReorg to use MedianTimePast

sipa committed:
Move fPayAtLeastCustomFee function to CC

jonasschnelli committed:
[Qt] use ASYMP_UTF8 (≈) whenever we show a fee that is not absolute

jonasschnelli committed:
[Qt] make use of the nMinimumTotalFee (absolute) in coincontrols fee calculation

Commit:
contrib: Del. gitian downloader config and update gitian README

jonasschnelli committed:
[Qt] improve minimum absolute fee option
- Only display the minimum absolute fee control if CoinControl is enabled

gmaxwell committed:
Add whitelistforcerelay to control forced relaying. [#7099 redux]

- Add whitelistforcerelay to control forced relaying.

Also renames whitelistalwaysrelay.

Nodes relay all transactions from whitelisted peers, this
 gets in the way of some useful reasons for whitelisting
 peers-- for example, bypassing bandwidth limitations.

The purpose of this forced relaying is for specialized gateway
 applications where a node is being used as a P2P connection
 filter and multiplexer, but where you don't want it getting
 in the way of (re-)broadcast.

This change makes it configurable with whitelistforcerelay.

- Blacklist -whitelistalwaysrelay; replaced by -whitelistrelay.

Github-Pull: #7439
Rebased-From: 325c725fb6205e38142914acb9ed1733d8482d46 89d113e02a83617b4e971c160d47551476dacc71

gmaxwell committed:
Replace setInventoryKnown with a rolling bloom filter.

Github-Pull: #7133
Rebased-From: ec73ef37eccfeda76de55c4ff93ea54d4e69e1ec e20672479ef7f2048c2e27494397641d47a4d88d 6b849350ab074a7ccb80ecbef387f59e1271ded6 b6a0da45db8d534e7a77d1cebe382cd5d83ba9b8 d41e44c9accb3df84e0abbc602cc76b72754d382 aa4b0c26b0a94ca6164c441aae723e118554d214

paveljanik committed:
Describe maxmempool and mempoolminfee in the getmempoolinfo RPC help.

laanwj committed:
Merge pull request #7112

9af5f9c Move uiInterface.NotifyBlockTip signal above the core/wallet signal - This will keep getbestblockhash more in sync with blocknotify callbacks (Jonas Schnelli)
4082e46 [Qt] call GuessVerificationProgress synchronous during core signal, pass double over UI signal (Jonas Schnelli)
947d20b [Qt] reduce cs_main in getVerificationProgress() (Jonas Schnelli)
e6d50fc [Qt] update block tip (height and date) without locking cs_main, update always (each block) (Jonas Schnelli)
012fc91 NotifyBlockTip signal: switch from hash (uint256) to CBlockIndex* - also adds a boolean for indication if the tip update was happening during initial sync - emit notification also during initial sync (Jonas Schnelli)

Commit:
Fix url in .travis.yml

laanwj committed:
Merge pull request #7035

4531fc4 torcontrol: only output disconnect if -debug=tor (Daniel Cousens)

laanwj committed:
Merge pull request #7008

c035306 Change GetPriority calculation. (Alex Morcos)
71f1d9f Modify variable names for entry height and priority (Alex Morcos)
5945819 Remove default arguments for CTxMemPoolEntry() (Alex Morcos)

laanwj committed:
Merge pull request #7094

1bb289f Assert now > 0 in GetTime GetTimeMillis GetTimeMicros (Patick Strateman)

Commit:
[qa] pull-tester: Cleanup (run keypool, tidy stdout)

* Run keypool (takes 6 seconds)
* Print duration of each rpc test
* Structure output (bold, new lines)

jonasschnelli committed:
[Tests] Add mempool_limit.py test

- [Tests] Add mempool_limit.py test
- [Tests] Refactor some shared functions

Github-Pull: #7153
Rebased-From: 110ff1142c5284edba8aab77fcac0bea0e551969 7632cf689a9b959dd7a059b8b4a04761a4bc6e6a

jonasschnelli committed:
add InMempool() function

sdaftuar committed:
Explicitly set chain limits in replace-by-fee test

gmaxwell committed:
Merge pull request #7044

d52fbf0 Added additional config option for multiple RPC users. (Gregory Sanders)

gmaxwell committed:
Merge pull request #7072

996d311 [RPC] Add transaction size to JSON output (Nick)

laanwj committed:
Merge pull request #7079

ebb25f4 Limit setAskFor and retire requested entries only when a getdata returns. (Gregory Maxwell)
5029698 prevent peer flooding request queue for an inv (kazcw)

laanwj committed:
Merge pull request #7105

9ac63d6 Keep track of explicit wallet conflicts instead of using mempool (Pieter Wuille)

laanwj committed:
rpc: Don't translate warning messages

But keep translating them in the GUI.
This - necessarily - requires duplication of a few messages.
Alternative take on #7134, that keeps the translations from being wiped.

Also document GetWarnings() input argument.

Fixes #5895.

laanwj committed:
Merge pull request #7135

fa3a38a [qa] pull-tester: Cleanup (run keypool, tidy stdout) (MarcoFalke)

laanwj committed:
Merge pull request #7096

ff723da [Qt] improve minimum absolute fee option - Only display the minimum absolute fee control if CoinControl is enabled (Jonas Schnelli)
31b508a [Qt] make use of the nMinimumTotalFee (absolute) in coincontrols fee calculation (Jonas Schnelli)
80462dd [Qt] use ASYMP_UTF8 (≈) whenever we show a fee that is not absolute (Jonas Schnelli)
ecc7c82 Move fPayAtLeastCustomFee function to CC (Pieter Wuille)

laanwj committed:
Merge pull request #6914

114b581 Prevector type (Pieter Wuille)

laanwj committed:
Merge pull request #7136

fa19a58 HelpMessage: Don't hide -mintxfee behind showDebug (MarcoFalke)
faffc17 rpcwallet: Clarify what settxfee does (MarcoFalke)
9999cb0 Fix url in .travis.yml (MarcoFalke)
fa22a10 contrib: Del. gitian downloader config and update gitian README (MarcoFalke)
fad3035 [doc] Minor markdown fixes (MarcoFalke)

laanwj committed:
Merge pull request #7063

2b31ab9 Add rpc test for prioritisetransaction (Suhas Daftuar)
6e8b07f Add rounding helper function to util.py (Suhas Daftuar)

laanwj committed:
qt: use QMetaObject::invokeMethod for cross-thread signaling in clientmodel

It's surprising to me that Q_EMIT even worked for this.

But it doesn't build in Qt4, so switch back to
`QMetaObject::invokeMethod`. Fixes #7138.

laanwj committed:
Merge pull request #7022

50947ef Change default block priority size to 0 (Alex Morcos)

laanwj committed:
Merge pull request #6915

2d8860e Fix removeForReorg to use MedianTimePast (Suhas Daftuar)
b7fa4aa Don't call removeForReorg if DisconnectTip fails (Suhas Daftuar)
7e49f5f Track coinbase spends in CTxMemPoolEntry (Suhas Daftuar)
bb8ea1f removeForReorg calls once-per-disconnect-> once-per-reorg (Matt Corallo)
474b84a Make indentation in ActivateBestChainStep readable (Matt Corallo)
b0a064c Fix comment in removeForReorg (Matt Corallo)
9b060e5 Fix removal of time-locked transactions during reorg (Matt Corallo)
0c9959a Add failing test checking timelocked-txn removal during reorg (Matt Corallo)

laanwj committed:
Merge pull request #7143

6da12df qt: use QMetaObject::invokeMethod for cross-thread signaling in clientmodel (Wladimir J. van der Laan)

laanwj committed:
Merge pull request #7141

aabc897 rpc: Don't translate warning messages (Wladimir J. van der Laan)

laanwj committed:
qt: periodic translations update

laanwj committed:
Merge pull request #7137

cfdc662 Explicitly set chain limits in replace-by-fee test (Suhas Daftuar)

laanwj committed:
test: Disable scheduler test manythreads

It causes occasional deadlocks, resulting in false negatives in Travis.

Disable the test for now.
Works around #6540.

paveljanik committed:
Name union to prevent compiler warning

sipa committed:
Merge pull request #6898

553cad9 Rewrite CreateNewBlock (Alex Morcos)
5f12263 Expose FormatStateMessage (Alex Morcos)
1f09287 Make accessing mempool parents and children public (Alex Morcos)
7230187 Add TxPriority class and comparator (Alex Morcos)
f3fe836 Add a score index to the mempool. (Alex Morcos)
c49d5bc Store the total sig op count of a tx. (Alex Morcos)

posita committed:
Add missing automake package to deb-based UNIX install instructions.

Rebased-From: b4404090259be4e34ef5dba33e47a41e7d9acc03
Github-Pull: #7152

jonasschnelli committed:
Merge pull request #7040

b171c69 [doc] Update OS X build notes for new qt5 configure (Michael Ford)

laanwj committed:
Merge pull request #6872

dd5862c Flush coins cache also after transaction processing (Pieter Wuille)
bde953e Uncache input txn in utxo cache if a tx is not accepted to mempool (Matt Corallo)
97bf377 Add CCoinsViewCache::HaveCoinsInCache to check if a tx is cached (Matt Corallo)
677aa3d Discard txn cache entries that were loaded for removed mempool txn (Matt Corallo)
b2e74bd Get the set of now-uncacheable-txn from CTxMemPool::TrimToSize (Matt Corallo)
74d0f90 Add method to remove a tx from CCoinsViewCache if it is unchanged (Matt Corallo)

laanwj committed:
Merge pull request #7144

8f0d79e test: Disable scheduler test manythreads (Wladimir J. van der Laan)

petertodd committed:
Remove old replace-by-fee tests

Made redundant by tests in qa/rpc-tests/replace-by-fee.py

Commit:
Squashed 'src/univalue/' changes from 5839ac3..2740c4f

2740c4f Merge branch '2015_11_escape_plan' into bitcoin
7482163 Add new testcase to Makefile.am
46098ee Version 1.0.1.
ccf3575 parser: Ensure multiple values cannot follow each other
eb6cd64 Omit Obj/Arr open token from jsonTokenIsValue() test
bfef9e2 Makefile.am: list recently added test data, fail{35,36}.json
3e319f3 parser: Tighten array, object syntax checks.
c74185c parser: transform C++ variables into bitmask
f2568bc Prefer C++ STL vector .at() for accessing object values.
8eafa26 travis: run parallel 'make distcheck'
fd448da test: Improve tester diagnostics.  Add failing test case from #15
2158205 Use internal, locale-independent isspace(), isdigit() implementations.
2ab9ad4 travis: Make 'make distcheck' for more comprehensive checks.
3339191 Escape all control characters

git-subtree-dir: src/univalue
git-subtree-split: 2740c4f71242086a7eb3dc32f812546ba9fad913

Commit:
Merge commit '982709199f1b4e9e35211c419a81938f9f1dd4ed' into bitcoin

laanwj committed:
Merge pull request #7155

092e9ad Remove old replace-by-fee tests (Peter Todd)

laanwj committed:
Merge pull request #7118

b212f94 Describe maxmempool and mempoolminfee in the getmempoolinfo RPC help. (Pavel Janík)

laanwj committed:
Merge pull request #7146

1812de9 Name union to prevent compiler warning (Pavel Janík)

laanwj committed:
Merge pull request #7147

9827091 Squashed 'src/univalue/' changes from 5839ac3..2740c4f (MarcoFalke)

laanwj committed:
Merge pull request #6216

db6047d Take the training wheels off anti-fee-sniping (Peter Todd)

laanwj committed:
qt: Final translation update before 0.12 fork

- Add new translations (finally, after a long time)
- update-translation script was not considering new translations - oops
  - fixed this, also remove (nearly) empty translations
- Update translation process, it was still describing the old repository
  structure

laanwj committed:
Merge pull request #7128

02354c9 Constrain rpcport default values to a single location in code (Luke Dashjr)

Commit:
Update contrib/devtools/README.md

* Fix order
* Update subtree check

Commit:
Fix typo in wallet.cpp

laanwj committed:
Merge pull request #7157

fabd10a Fix typo in wallet.cpp (MarcoFalke)
fad2460 Update contrib/devtools/README.md (MarcoFalke)
5e151a8 PartitionCheck: remove useless spaces (paveljanik)
fad0088 TRIVIAL: Chainparams: Remove unused CBaseUnitTestParams (Jorge Timón)
74f7341 Update miner.cpp: Fix typo in comment (antonio-fr)
e69bad1 [trivial] Fix typo in peertablemodel.cpp (MarcoFalke)
8a03727 Fix various typos (paveljanik)

Commit:
[qt] Use tr() instead of _()

Also, `make translate`

Commit:
[qa] Fix pyton syntax in rpc tests

Github-Pull: #7335
Rebased-From: 7777994846cdb9b9cf69e391a33eeed30393bbcf

laanwj committed:
Merge pull request #7158

fab8347 [qt] Use tr() instead of _() (MarcoFalke)

laanwj committed:
Bump version to 0.12.0

Bump version from 0.11.99 to 0.12.0 so that we don't forget to do this
when rc1 is released.

gmaxwell committed:
Disconnect on mempool requests from peers when over the upload limit.

Mempool requests use a fair amount of bandwidth when the mempool is large,
 disconnecting peers using them follows the same logic as disconnecting
 peers fetching historical blocks.

Rebased-From: 6aadc7557823b7673b8f661b3d41cf867e2936a3
Github-Pull: #7166

laanwj committed:
test: remove necessity to call create_callback_map

Remove necessity to call create_callback_map (as well as the function
itself) from the Python P2P test framework. Invoke the appropriate
methods directly.

- Easy to forget to call it and wonder why it doesn't work
- Simplifies the code
- This makes it easier to handle new messages in subclasses

Github-Pull: #7171
Rebased-From: 2f601d215da1683ae99ab9973219044c32fa2093

laanwj committed:
tests: Disable Tor interaction

This is unnecessary during the current tests (any test for Tor
interaction can explicitly enable it) and interferes with the proxy
test.

Github-Pull: #7170
Rebased-From: 4c40ec0451a8f279f3e2e40af068c9451afd699e

TheBlueMatt committed:
Don't do mempool lookups for "mempool" command without a filter

Github-Pull: #7174
Rebased-From: 96918a2f0990a8207d7631b8de73af8ae5d24aeb

laanwj committed:
net: Fix sent reject messages for blocks and transactions

Ever since we #5913 have been sending invalid reject messages
for transactions and blocks.

test: Add basic test for `reject` code

Extend P2P test framework to make it possible to expect reject
codes for transactions and blocks.

Github-Pull: #7179
Rebased-From: 9fc6ed6003da42f035309240c715ce0fd063ec03 20411903d7b356ebb174df2daad1dcd5d6117f79

laanwj committed:
net: Add and document network messages in protocol.h

- Avoids string typos (by making the compiler check)
- Makes it easier to grep for handling/generation of a certain message type
- Refer directly to documentation by following the symbol in IDE
- Move list of valid message types to protocol.cpp:
    protocol.cpp is a more appropriate place for this, and having
    the array there makes it easier to keep things consistent.

Github-Pull: #7181
Rebased-From: 9bbe71b641e2fc985daf127988a14a67c99da50a

pstratem committed:
Note that reviewers should mention the commit hash of the commits they reviewed.

Github-Pull: #7185
Rebased-From: e1030dddab11553d2854c1f466e5757d9815bfb8

Commit:
[wallet] Adjust pruning test

Github-Pull: #7193
Rebased-From: fafd09375eb5133abf921132643384a1ac6fa444

jamesob committed:
Fix help, add RPC tests for getblockheader

- Add assert_is_hex_string and assert_is_hash_string to RPC test utils.
- Add RPC documentation for getblockheader[chainwork].
- Add RPC tests for getblockheader.

Github-Pull: #7194
Rebased-From: 16d4fce0b203bdaa679ad5b3f1e6b6f46880d5d2 4745636126d9a4f28f701f701be392779815a7bf 135d6ec8cedc83ad800da45080c16d49e9182e80

SuriyaaKudoIsc committed:
Set link from http:// to https://

For opensource.org/licenses/MIT!
Github-Pull: #7197
Rebased-From: 00423e1a71204696ec37e6d757f9afe091bc7ee1

accraze committed:
Checks for null data transaction before issuing error to debug.log

CWalletTx::GetAmounts could not find output address for null data transactions, thus issuing an error in debug.log. This change checks to see if the transaction is OP_RETURN before issuing error.

resolves #6142

Github-Pull: #7200
Rebased-From: b6915b82398d2e1d1f888b3816adfaf06d9a450e c611acc38a95d336a824b632823aa1b652e570df d812daf967ba4173bfa1c37eeb4ab7a0ccc4df25

Commit:
Bump copyright headers to 2015

- Bump copyright headers to 2015
- [devtools] Rewrite fix-copyright-headers.py
- [devtools] Use git pretty-format for year parsing

Github-Pull: #7205
Rebased-From: fa6ad855e9159b2247da4fa0054f32fa181499ab fa24439ff3d8ab5b9efaf66ef4dae6713b88cb35 fa71669452e57039e4270fd2b33a0e0e1635b813

TheBlueMatt committed:
Add "NODE_BLOOM" to guiutil so that peers don't get UNKNOWN[4]

Rebased-From: daf6466330d9d3e4d9034fd316cded192d2a7d67
Github-Pull: #7206

jonasschnelli committed:
[RPC-Tests] add option to run rpc test over QT clients

Github-Pull: #7068
Rebased-From: 979698c1715ce86a98934e48acadbc936c95c9a3

laanwj committed:
test: don't override BITCOIND and BITCOINCLI if they're set

In rpc-tests.py, don't override BITCOIND and BITCOINCLI if they're
already set. Makes it possible to run the tests with either another tree
or the GUI.

Github-Pull: #7209
Rebased-From: 83cdcbdca41583a5a754a89f45b04b56cd0df627

Commit:
Rename OP_NOP2 to OP_CHECKLOCKTIMEVERIFY.

Github-Pull: #7213
Rebased-From: 37d271d7cce44885f835292ffe99b54399b014d6

Commit:
qt5: Use the fixed font the system recommends

Github-Pull: #7214
Rebased-From: fa2f4bc4eb0f21f5be8c88954ae2d99c5b18b987

Commit:
Removed offline testnet DNSSeed 'alexykot.me'.

Github-Pull: #7216
Rebased-From: e18378e53fb71c39236db35ab2d560b43602b1be

Commit:
torcontrol debug: Change to a blanket message that covers both cases

Commit:
[qa] wallet: Check if maintenance changes the balance

- [qa] Cleanup wallet.py test
- [qa] check if wallet or blochchain maintenance changes the balance
- [walletdb] Add missing LOCK() in Recover() for dummyWallet

Github-Pull: #7229
Rebased-From: fa0765d433eb6d44a5cbec44f136b62814c663e5 fa14d994843fe2d700c977653cd3133d0a77cb67 fa33d9740c9b0d1071094ab6c1736f27a7090c95

sdaftuar committed:
Mark blocks with too many sigops as failed

Github-Pull: #7217
Rebased-From: 5246180f168c9b761b6158b0725f5718239ba66c

sdaftuar committed:
Add more tests to p2p-fullblocktest

Github-Pull: #7226
Rebased-From: 9b41a5fba278e9ab56a9b86e7a5fe195dcad0b7a

sdaftuar committed:
RPC: indicate which transactions are replaceable

Add "bip125-replaceable" output field to listtransactions and gettransaction
which indicates if an unconfirmed transaction, or any unconfirmed parent, is
signaling opt-in RBF according to BIP 125.

Github-Pull: #7286
Rebased-From: eaa8d2754b48b62cdd07255fc3028feecad0c095

instagibbs committed:
Added help text for chainwork value

Github-Pull: #7232
Rebased-From: 94bdd71f9b4768c9803ffd133aa7781b19bfa6f9

Commit:
[qa] Move gen_return_txouts() to util.py

Github-Pull: #7250
Rebased-From: fa0a9749eb09f6b537b98075241a7fcb46f758e3

Commit:
[gitian] Set reference date to something more recent

Github-Pull: #7251
Rebased-From: fa095622c25492ddc7096c5825f327e4427e7d75

Commit:
[doc] Update release-process.md

Conflicts:
	doc/release-process.md

Github-Pull: #7465
Rebased-From: fa616c2fedd19d8e88f042abd5e99ac9595923df

luke-jr committed:
Bugfix: update-translations: Allow numerus translations to omit %n specifier (usually when it only has one possible value)

Github-Pull: #7253
Rebased-From: 0d595894f028248a1a1b00491dad95320844c685

fanquake committed:
Replace some instances of formatWithUnit with formatHtmlWithUnit

Strings in a HTML context should be using formatHtmlWithUnit.

Github-Pull: #7255
Rebased-From: 5fdf32de7ed85a1a0aec7cdedb83f750f4a0f7ff

Commit:
[qt] Intro: Display required space

Required space depends on the user's choice:
 -prune=0
 -prune=<n>

Commit:
[debian] Bump manpages and only mention -?

The manpages are outdated and are very rarely updated when changes
to the code happen.

Github-Pull: #7274
Rebased-From: fae7a369cb137000897d32afab7eb13aa79ec34e fa6ce44bf98fe1dd5be779fd77b844e7016d209e

sipa committed:
Report non-mandatory script failures correctly

Github-Pull: #7276
Rebased-From: 7ef8f3c072a8750c72a3a1cdc727b5c1d173bac8

laanwj committed:
Merge pull request #7245

fa8c8d7 torcontrol debug: Change to a blanket message that covers both cases (MarcoFalke)

jonasschnelli committed:
[Qt] fix coincontrol update issue when deleting a send coin entry

Github-Pull: #7282
Rebased-From: 621bd6919f47be4d23091d8ae7c980f9567d83a9

laanwj committed:
qt: periodic translations pull from transifex

laanwj committed:
Update translations after #7253

Include translations that omit raw number from the singular case.

Commit:
[init] Add missing help for args

Github-Pull: #7290
Rebased-From: fa6ab96799f9d7946200fb646fefe35c6daab9b2 faa572a3296c0955dcb2cc0bd9b925c2a31e7892 fa461df685063e6b12664fe6928362484f690f01

Commit:
[init] Fix error message of maxtxfee invalid amount

Github-Pull: #7290
Rebased-From: fac11ea3106ff29ec884dfe9d9b287fd1beda5fc

Commit:
[wallet] Add regression test for vValue sort order

- [wallet] Add regression test for vValue sort order
- [trivial] Merge test cases and replace CENT with COIN

Github-Pull: #7293
Rebased-From: fa3c7e644f427329bcffa1a5600fdbd7e97c837f faf538bfdbb4ecebde73e95c80718c2d9ecee1f5

morcos committed:
Add sane fallback for fee estimation

- Always respect GetRequiredFee for wallet txs
- Add sane fallback for fee estimation
- SQUASHME: Fix rpc tests that assumed fallback to minRelayTxFee

Add new commandline option "-fallbackfee" to use when fee estimation does not have sufficient data.

Github-Pull: #7296
Rebased-From: 995b9f3 e420a1b bebe58b

morcos committed:
Make sure conflicted wallet tx's update balances

Github-Pull: #7306
Rebased-From: f61766b37beb2fecbe3915a72a814cbdb107be0a

sdaftuar committed:
[Tests] Eliminate intermittent failures in sendheaders.py

- Add race-condition debugging tool to mininode
- Eliminate race condition in sendheaders.py test

Clear the last block announcement before mining new blocks.

Github-Pull: #7308
Rebased-From: 82a0ce09b45ab9c09ce4f516be5b9b413dcec470 168915e6dec88b31793d4ee4b60b94d4149de36c

Commit:
Update release-notes.md

Transaction memory pool limiting

Priority transactions

Wallet transaction fees

Commit:
Expand section "Wallet transaction fees" & fix format and typos

morcos committed:
Add RPC call abandontransaction

- Make wallet descendant searching more efficient
- Add new rpc call: abandontransaction

Unconfirmed transactions that are not in your mempool either due to eviction or other means may be unlikely to be mined.  abandontransaction gives the wallet a way to no longer consider as spent the coins that are inputs to such a transaction.  All dependent transactions in the wallet will also be marked as abandoned.

- Add RPC test for abandoned and conflicted transactions.
- [Wallet] Call notification signal when a transaction is abandoned

Github-Pull: #7312
Rebased-From: 9e697172542e2b01517e4025df2c23d0ed5447f4 01e06d1fa365cedb7f5d5e17e6bdf0b526e700c5 df0e2226d998483d247c0245170f6b8ff6433b1d d11fc1695c0453ef22a633e516726f82717dd1d9

jonasschnelli committed:
Backport: quickfix for RPC timer interface problem

laanwj committed:
Merge pull request #7319

b1a8374 [qt] Intro: Display required space (MarcoFalke)

laanwj committed:
Merge pull request #7318

daa8da2 Backport: quickfix for RPC timer interface problem (Jonas Schnelli)

luke-jr committed:
Merge branch bytespersigop

laanwj committed:
Merge pull request #7323

45b8e27 -bytespersigop option to additionally limit sigops in transactions we relay and mine (Luke Dashjr)

luke-jr committed:
doc/bips: Document BIP 125 support

Commit:
Add Replace-by-fee to release-notes

laanwj committed:
Merge pull request #7324

Commit:
Add fallbackfee default value

Commit:
[wallet] Clarify rpc help message with regard to rounding

laanwj committed:
Merge pull request #7309

fab88af Add fallbackfee default value (MarcoFalke)
fa0a391 Add Replace-by-fee to release-notes (MarcoFalke)
fa4ba40 Expand section "Wallet transaction fees" & fix format and typos (MarcoFalke)
fabba1c Update release-notes.md (MarcoFalke)

laanwj committed:
Preliminary release notes 0.12.0

laanwj committed:
Update translations pre-rc1

laanwj committed:
Merge pull request #7332

laanwj committed:
Merge pull request #7333

a06a8b4 add InMempool() function (Jonas Schnelli)

laanwj committed:
doc: Remove BIP65 mention from release notes

This is already done, not new in 0.12.

crowning- committed:
[Wallet] Transaction View: LastMonth calculation fixed

Github-Pull: #7327
Rebased-From: 30cdacea3c356acda32ab77238f07c1c40b1f1b5

morcos committed:
release note fixups

luke-jr committed:
release-notes: Combine NOP2->CLTV asm change into "RPC: Low-level API changes" section

theuni committed:
release: remove libc6 dependency from the osx signing descriptor

It is unneeded after the last toolchain update, and missing from Trusty.

Github-Pull: #7342
Rebased-From: 3503a78670d0eacf39c618b45b08581dfb3ed68f

Commit:
[doc] backwards-compatibility issues due to chainstate obfuscation

morcos committed:
Add comment about mining changes and more about priority

luke-jr committed:
release-notes: Cover priority changes correctly, removing mentions of possible futures

sdaftuar committed:
Eliminate race condition in mempool_packages test

Github-Pull: #7368
Rebased-From: 4d10d2e16fb837abe304e0a5d3bc0a41941d917a

btcdrak committed:
Update project URL

Github-Pull: #7328
Rebased-From: b07b103e8af14cd3fca44dca7bc694d2c3bffcc1

laanwj committed:
Merge pull request #7336

82667af release note fixups (Alex Morcos)

Commit:
[qt] Windows: Make rpcconsole monospace font larger

Github-Pull: #7364
Rebased-From: fa6a59dd397e62e850fc57df05cd6d117fbdcd82

prayagverma committed:
Update license year range to 2016

Conflicts:
	configure.ac

Github-Pull: #7363
Rebased-From: bd34174ebca239e6796f0eb2015ddc2f218aac3c

sdaftuar committed:
Mention mempool chain limits in release notes

laanwj committed:
Merge pull request #7367

2e552b0 Mention mempool chain limits in release notes (Suhas Daftuar)

laanwj committed:
Merge pull request #7347

5cacb8f Add comment about mining changes and more about priority (Alex Morcos)

laanwj committed:
Merge pull request #7338

f17b00b release-notes: Combine NOP2->CLTV asm change into "RPC: Low-level API changes" section (Luke Dashjr)
e20704b Replace some instances of formatWithUnit with formatHtmlWithUnit (fanquake)
6f8346d qt5: Use the fixed font the system recommends (MarcoFalke)
605de4a Rename OP_NOP2 to OP_CHECKLOCKTIMEVERIFY. (mb300sd)
6191a9b [RPC-Tests] add option to run rpc test over QT clients (Jonas Schnelli)
6307beb Note that reviewers should mention the commit hash of the commits they reviewed. (Patrick Strateman)
6092ff2 Set link from http:// to https:// (Suriyaa Kudo)

laanwj committed:
Merge pull request #7345

fa8c497 [doc] backwards-compatibility issues due to chainstate obfuscation (MarcoFalke)

Commit:
[Doc] Wallet & Pruning

sdaftuar committed:
Fix error in blockchain.py introduced in merge

Github-Pull: #7373
Rebased-From: 4a0487937877484f14476716c3643de7a31c32da

laanwj committed:
Merge pull request #7370

fa31133 [Doc] Wallet & Pruning (MarcoFalke)

gavinandresen committed:
Minimal consensus/miner changes for 2mb block size bump.

Conflicts:
	src/main.h
	src/miner.cpp

Commit:
[walletdb] Fix syntax error in key parser

Github-Pull: #7381
Rebased-From: fa6d4cc09575de30386bfbc5c8c3858cd7a2f42a

Commit:
[qt] Peertable: Increase SUBVERSION_COLUMN_WIDTH

Github-Pull: #7384
Rebased-From: faa9011d09d7429b97ec7595f9f77abf8ea770d3

laanwj committed:
Add option `-permitrbf` to set transaction replacement policy

Add a configuration option `-permitrbf` to set transaction replacement policy
for the mempool.

Enabling it will enable (opt-in) RBF, disabling it will refuse all
conflicting transactions.

Conflicts:
	src/init.cpp
	src/main.cpp
	src/main.h

Github-Pull: #7386
Rebased-From: b768108d9c0b83330572711aef1e569543130d5e

sipa committed:
Get rid of inaccurate ScriptSigArgsExpected

sdaftuar committed:
doc: Update release notes for 0.12

Update and reword BIP 125 section

Mention changes to banlist (clearbanned/setban)

Pruning nodes can relay

gavinandresen committed:
Two megabyte fork after miner vote and grace period

Activation code for a two megabyte block size increase.

Activation condition:

750 of 1000 blocks with version bit 0x10000000 set, then a four-week
(28-day) grace period, after which miners may create 2MB
blocks.

Conflicts:
	qa/pull-tester/rpc-tests.sh
	qa/rpc-tests/bipdersig-p2p.py
	src/chainparams.cpp
	src/main.cpp
	src/miner.cpp

Additionall; Fix unit test.
Instead of relying on default, specify block size.

laanwj committed:
Merge #7387: Get rid of inaccurate ScriptSigArgsExpected

52b29dc Get rid of inaccurate ScriptSigArgsExpected (Pieter Wuille)

laanwj committed:
Merge #7371: [0.12] backports

236686b [init] Add missing help for args (MarcoFalke)
44438a1 [init] Fix error message of maxtxfee invalid amount (MarcoFalke)
a74fa1f [Wallet] Transaction View: LastMonth calculation fixed (crowning-)

laanwj committed:
qt: pre-rc2 translations update

laanwj committed:
doc: Add commits since rc1 to release notes

laanwj committed:
doc: forgot #7222 in release notes

gavinandresen committed:
Accurate sigop/sighash accounting and limits

Adds a ValidationCostTracker class that is passed to
CheckInputs() / CScriptCheck() to keep track of the exact number
of signature operations required to validate a block, and the
exact number of bytes hashed to compute signature hashes.

Also extends CHashWriter to keep track of number of bytes hashed.

Signature operations per block are limited to MAX_BLOCK_SIGOPS
(unchanged at 20,000)

Bytes hashed to compute signatures is limited to MAX_BLOCK_SIGHASH
(1.3 GB in this commit).

Conflicts:
	src/main.cpp
	src/miner.cpp
	src/script/interpreter.h

rusty-loy committed:
Rename Core to Classic

Squased commits from rustyloy and others and fixed merge
conflicts.

laanwj committed:
net: Hardcoded seeds update January 2016

Github-Pull: #7415
Rebased-From: 4818dba90074f213efa0fa7faf577ce5fb02eaee

pdpi committed:
Fix translation path
Conflicts:
	.tx/config

xor-freenet committed:
doc: Explain effects of -prune=<n> parameter in release notes

As discussed in the mailing list thread:
    [bitcoin-dev] Bitcoin Core 0.12.0 release candidate 1 available
in the replies to this message:
http://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-January/012276.html

Please review thoroughly, I'm a newbie.

xor-freenet committed:
doc: Fix minimal disk usage with pruning enabled

xor-freenet committed:
doc: In release notes, do not claim that pruning is leeching

Peter Todd says it does not matter currently.

xor-freenet committed:
doc: In release notes, increase estimate of disk usage with pruning

Jonas Schnelli recommended this to account for growth of utxo set, debug
log, etc.

btcdrak committed:
Update release-notes.md

theuni committed:
release: Add security/export checks to gitian and fix current failures

- fix parsing of BIND_NOW with older readelf
- add _IO_stdin_used to ignored exports

For details see: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=634261#109

- add check-symbols and check-security make targets

These are not added to the default checks because some of them depend on
release-build configs.

- always link librt for glibc back-compat builds

glibc absorbed clock_gettime in 2.17. librt (its previous location) is safe to
link in anyway for back-compat.

Fixes #7420

- add security/symbol checks to gitian

Github-Pull: #7424
Rebased-From: cd27bf51e06a8d79790a631696355bd05751b0aa 475813ba5b208eb9a5d027eb628a717cc123ef4f f3d3eaf78eb51238d799d8f20a585550d1567719 a8ce872118c4807465629aecb9e4f3d72d999ccb a81c87fafce43e49cc2307947e3951b84be7ca9a

laanwj committed:
Merge #7421: [doc] Release notes update for 0.12

65d384f doc: Update release notes for 0.12 (Suhas Daftuar)

laanwj committed:
Merge #7422: Improve section title in release-notes.md

42b521d Update release-notes.md (฿tcDrak)

xor-freenet committed:
doc: Minor sentence length / capitalization fixes

xor-freenet committed:
doc: Fix wrong claims about blockchain reorganization with pruning

xor-freenet committed:
doc: Minor spelling fix

xor-freenet committed:
doc: In release notes, reduce length of pruning section

laanwj committed:
Merge #7416: doc: Explain effects of -prune=<n> parameter in release notes

8c5f903 doc: In release notes, reduce length of pruning section (xor-freenet)
15c0263 doc: Minor spelling fix (xor-freenet)
54d3907 doc: Fix wrong claims about blockchain reorganization with pruning (xor-freenet)
a683d20 doc: Minor sentence length / capitalization fixes (xor-freenet)
58e3abf doc: In release notes, increase estimate of disk usage with pruning (xor-freenet)
be4b474 doc: In release notes, do not claim that pruning is leeching (xor-freenet)
46d7eb6 doc: Fix minimal disk usage with pruning enabled (xor-freenet)
fe074cc doc: Explain effects of -prune=<n> parameter in release notes (xor-freenet)

luke-jr committed:
Rename permitrbf to replacebyfee

"permit" is currently used to configure transaction filtering, whereas replacement is more to do with the memory pool state than the transaction itself.

luke-jr committed:
Accept replacebyfee=opt-in for turning on opt-in RBF

Basic forward-compatibility with more flexible parameters like fss

gmaxwell committed:
Do not absolutely protect local peers from eviction.

With automatic tor HS support in place we should probably not be providing
 absolute protection for local peers, since HS inbound could be used to
 attack pretty easily.  Instead, this counts on the latency metric inside
 AttemptToEvictConnection to privilege actually local peers.

jgarzik committed:
Update gitian descriptors for classic.

Credit: pdpi
Fixes #38

luke-jr committed:
Simplify check for replacebyfee=opt-in

luke-jr committed:
release-notes: Update for permitrbf->replacebyfee rename

laanwj committed:
Merge #7438: Do not absolutely protect local peers; decide group ties based on time.

8e09f91 Decide eviction group ties based on time. (Gregory Maxwell)
46dbcd4 Do not absolutely protect local peers from eviction. (Gregory Maxwell)

luke-jr committed:
Rename replacebyfee=opt-in to mempoolreplacement=fee

luke-jr committed:
release-notes: Update for replacebyfee->mempoolreplacement rename

gavinandresen committed:
Do not relay or mine excessive sighash transactions

This is a belt-and-suspenders fix to make sure CreateNewBlock() or external mining software can never produce a block that violates the MAX_BLOCK_SIGHASH rule.

It does this by rejecting transactions that do too much signature hashing -- they are not added to the memory pool, and so will not be considered for inclusion in new blocks.

How the code works: every transaction uses up some fraction of the MAX_BLOCK_SIZE limit and the MAX_BLOCK_SIGHASH limit. If a transaction uses up a larger fraction of the SIGHASH limit than the SIZE limit, it is rejected. That ensures that no matter which transactions are selected for the block, the SIZE limit will be hit before the SIGHASH limit.

This is a much simpler solution than modifying CreateNewBlock or external transaction selection software to keep track of the SIZE limit, the SIGOPS limit, AND the new SIGHASH limit.

This is belt-and-suspenders because, in practice, the 100,000-byte IsStandard size limit prevents the block SIGHASH limit from being hit.

The IsStandard code related to the old SIGOPS limit is left unchanged.

Conflicts:
	src/main.cpp

laanwj committed:
Merge #7440: [0.12] Rename permitrbf to mempoolreplacement and provide minimal string-list forward compatibility

af9f564 release-notes: Update for replacebyfee->mempoolreplacement rename (Luke Dashjr)
4ad418b Rename replacebyfee=opt-in to mempoolreplacement=fee (Luke Dashjr)
b2287a7 release-notes: Update for permitrbf->replacebyfee rename (Luke Dashjr)
5f456a6 Simplify check for replacebyfee=opt-in (Luke Dashjr)
e8d19ab Accept replacebyfee=opt-in for turning on opt-in RBF (Luke Dashjr)
1205f87 Rename permitrbf to replacebyfee (Luke Dashjr)

laanwj committed:
doc: update release notes for rc3

laanwj committed:
qt: translations update pre-rc3

zander committed:
Make internal (core) errors show up in the Qt client.

jtoomim commented on src/alert.h:
```diff
@@ -46,7 +46,7 @@ class CUnsignedAlert
     std::string strReserved;
 
-    ADD_SERIALIZE_METHODS;
+    ADD_SERIALIZE_METHODS
```
Style nit: 

ADD_SERIALIZE_METHODS is a macro which substitutes in a self-sufficient statement or block with its own ending semicolon or {} pair, and which therefore does not need the trailing semicolon here? If so, then the semicolon that you removed has no use to the compiler, but it does help with readability, as it indicates to the reader that the ADD_SERIALIZE_METHODS macro does not syntactically interact with the template inline void SerializationOp method below. 

Other files (wallet.h, walletdb.h, crypter.h, undo.h) also have ADD_SERIALIZE_METHODS;, and use the semicolon. I think it would be better to not make modifications to that stylistic element in one place that is inconsistent with other instances of the same idiom, and that we also shouldn't change lines without good reason for the sake of mergeability of pulls from Core.

zander replied:
Its not a style nit because its actually a warning from both my IDE and I'm sure with the right flags, from g++

gavinandresen commented:
Three things:

How did you test?

You need to LOCK(cs_mapAlerts) before modifying mapAlerts.

And I think CAlert::Notify() is now used only by CAlert::ProcessAlert/NotifyInternal, so it can be made private to CAlert.  In fact, thinking about it, maybe it'd be better to make NotifyInternal the private routine the ProcessAlert and Notify both call...

Also: I agree with @jtoomim on the extra ; warning-- better to stay consistent with Core at this point.

zander commented:
> Three things:
> 
> How did you test?

Added these lines to the bottom of updateTip()

```
strMiscWarning = "Warning: This version is too new; downgrade required!";
CAlert::NotifyInternal(strMiscWarning);
```

> You need to LOCK(cs_mapAlerts) before modifying mapAlerts.
> Good point.
> 
> And I think CAlert::Notify() is now used only by
> CAlert::ProcessAlert/NotifyInternal, so it can be made private to CAlert. 
> In fact, thinking about it, maybe it'd be better to make NotifyInternal the
> private routine the ProcessAlert and Notify both call...

We could make it private, sure.

> Also: I agree with jtoomim on the extra ; warning-- better to stay
> consistent with Core at this point.

Ok, no problem. See updated commit.

gavinandresen commented on src/alert.cpp:
```diff
@@ -244,8 +246,20 @@ bool CAlert::ProcessAlert(const std::vector<unsigned char>& alertKey, bool fThre
+    static CAlert internalAlert;
+    internalAlert.vchMsg = std::vector<unsigned char>(strMessage.begin(), strMessage.end());
+    uint256 zero;
+    LOCK(cs_mapAlerts);
```
You should limit the scope of the LOCK to just the next line, so cs_mapAlerts is not locked when NotifyAlertChanged / Notify is called.

jtoomim commented on src/alert.cpp:
```diff
@@ -244,8 +246,22 @@ bool CAlert::ProcessAlert(const std::vector<unsigned char>& alertKey, bool fThre
     return true;
 }
 
```
Using zero as the key sounds a bit sketchy, as we would likely get the second error erasing or replacing the first. Perhaps you could hash the strMessage and use the hash instead of zero? Or was overwriting the messages intended behavior? Or is there something I'm missing?

gavinandresen replied:
If I recall correctly, the GUI will only show one alert at a time (the highest priority alert).

So using one static CAlert instance with a zero hash for all internally-generated alerts is the Simplest Thing That Will Work, and should work just fine.

Tom or Jonathan: you could do a quick test by adding some code in init.cpp that calls CAlert::NotifyInternal after a little bit.  Something like:

```
scheduler.scheduleFromNow(boost::bind(CAlert::NotifyInternal, "One Minute Warning!"), 60);
scheduler.scheduleFromNow(boost::bind(CAlert::NotifyInternal, "Two Minute Warning!"), 120);
```

zander commented:
You recall correctly, Gavin; I tested this when I ran the Core code which shows a notification saying its pre-release and after some time my test code replaced that message with my own.

It, in short, only shows one.
Also, for those that don't use the GUI very often, it is good to know that this is not a dialog, it is instead a permanent coloured text overlaying the main UI. Not entirely intrusive, yet hard to ignore.

Vlad pointed out that this means the warnings can never be removed, which is an issue indeed.
It feels out of scope of this patch to fix that, though.

saltduck committed:
pass the travis-ci build

Use legacy group instead of default precise group.

The precise group does not support IPv6.

pdpi committed:
Fix autoconf reference to github repo.

ojanssens committed:
Create oj.pgp

Cherry-pick of 656ce711cf

zander committed:
Add public key

zander committed:
Make travis happy

pdpi committed:
Add public key

Cherry-pick of de875808b1

keo committed:
Travis: fix MacOSX SDK error

mrbandrews committed:
[rpc-tests] Change solve() to use rehash

Github-Pull: #7468
Rebased-From: 7689041c03278a09c88a2bb78cd00217f6d4b1de

pastly committed:
Fix spelling: misbeha{b,v}ing

Github-Pull: #7469
Rebased-From: 0830552673e37142599de897e87510f2f9866e1e

laanwj committed:
rpc: Add WWW-Authenticate header to 401 response

A WWW-Authenticate header must be present in the 401
response to make clients know that they can authenticate,
and how.

    WWW-Authenticate: Basic realm="jsonrpc"

Fixes #7462.

Github-Pull: #7472
Rebased-From: 7c06fbd8f58058d77c3e9da841811201d2e45e92

laanwj committed:
doc: add PR authors to release notes

Take full names from github API if available, otherwise
github username.

instagibbs committed:
Changed getnetworkhps value to double to avoid overflow.

Github-Pull; #7480
Rebased-From: 993d089e82fc045d7b0f23e1a5dc934cba0e3306

sdaftuar committed:
Update nQueuedValidatedHeaders after peer disconnection

Github-Pull: #7482
Rebased-From: 301bc7bc7e83f4c268c1722558b07dbb5b55fa92

luke-jr committed:
release-notes: Mention possibility of priority removal in 0.13, uncertainty of priority calculation being changed back, and request community input

laanwj committed:
doc: Update release notes for rc4 changes

laanwj committed:
qt: Translations update pre-rc4

gavinandresen committed:
Point at correct detached signatures repository

laanwj committed:
wallet: Ignore MarkConflict if block hash is not known

If number of conflict confirms cannot be determined, this means
that the block is still unknown or not yet part of the main chain,
for example during a reindex. Do nothing in that case,
instead of crash with an assertion.

Fixes #7234.

Github-Pull: #7491
Rebased-From: 40e7b61835cbe5fd471d0b4b71972526bf0e523c

luke-jr committed:
release-notes: Remove suggestion to use 0.11

gmaxwell committed:
release-notes: Significantly rewrite priority transactions section

luke-jr committed:
release-notes: Minor corrections and clarifications re Priority

sipa committed:
Correctly report high-S violations

Github-Pull: #7500
Rebased-From: 9d95187d5ddee56b6dfb55985008bdf70aed31f2

sipa committed:
Update the wallet best block marker when pruning

Github-Pull: #7502
Rebased-From: e4eebb604e19f67b0c7a483b1ded1229d75ecdd3

laanwj committed:
doc: Release notes update pre-rc5

laanwj committed:
qt: Translation update pre-rc5

laanwj committed:
doc: fix author list in release notes

laanwj committed:
Merge #7346: 0.12 release notes: Mining Policy Changes

b460004 release-notes: Minor corrections and clarifications re Priority (Luke Dashjr)
3450f4c release-notes: Significantly rewrite priority transactions section (Gregory Maxwell)
d0dbb6d release-notes: Remove suggestion to use 0.11 (Luke Dashjr)
73a0375 release-notes: Mention possibility of priority removal in 0.13, uncertainty of priority calculation being changed back, and request community input (Luke Dashjr)
4b8d2bc release-notes: Cover priority changes correctly, removing mentions of possible futures (Luke Dashjr)

zander committed:
Report Classic in p2p version message

zander committed:
Update COPYING
